tulip/synth.py

```diff
@@ -980,28 +980,42 @@ def synthesize_many(specs, ts=None, ignore_init=None,
     assert isinstance(ts, dict)
     for name, t in ts.items():
         assert isinstance(t, transys.FiniteTransitionSystem)
-        ignore = name in ignore_init
-        bool_act = name in bool_actions
+        ignore = ignore_init.get(name) 
+        bool_act = bool_actions.get(name)
         statevar = name
         if t.owner == 'sys':
-            specs |= sys_to_spec(t, ignore, statevar,
+            sys_spec = sys_to_spec(t, ignore, statevar,
                                  bool_actions=bool_act)
+            sys_spec.qinit = specs.qinit 
+            sys_spec.moore = specs.moore 
+            sys_spec.plus_one = specs.plus_one 
+            specs |= sys_spec
         elif t.owner == 'env':
-            specs |= env_to_spec(t, ignore, statevar,
+            env_spec = env_to_spec(t, ignore, statevar,
                                  bool_actions=bool_act)
+            env_spec.qinit = specs.qinit 
+            env_spec.moore = specs.moore 
+            env_spec.plus_one = specs.plus_one 
+            specs |= env_spec
```

Comment:
Similar comments apply here, with reference to the relevant area:

https://github.com/tulip-control/tulip-control/blob/1ad5b12ce1f96253d898abb8485d4e0ca8d20d90/tulip/synth.py#L1236-L1239